libs/ngxtension/inject-local-storage/src/inject-local-storage.ts

```diff
@@ -0,0 +1,113 @@
+import {
+	effect,
+	inject,
+	InjectionToken,
+	signal,
+	type WritableSignal,
+} from '@angular/core';
+
+export const NGXTENSION_LOCAL_STORAGE = new InjectionToken(
+	'NGXTENSION_LOCAL_STORAGE',
+	{
+		providedIn: 'root',
+		factory: () => localStorage, // this would be the default
+	},
+);
+
+export function provideLocalStorageImpl(impl: typeof globalThis.localStorage) {
+	return {
+		provide: NGXTENSION_LOCAL_STORAGE,
+		useValue: impl,
+	};
+}
+
+/**
+ * Options to override the default behavior of the local storage signal.
+ */
+export type LocalStorageOptions<T> = {
+	/**
+	 * The default value to use when the key is not present in local storage.
+	 */
+	defaultValue?: T | (() => T);
+	/**
+	 *
+	 * Determines if local storage syncs with the signal.
+	 * When true, updates in one tab reflect in others, ideal for shared-state apps.
+	 * Defaults to true.
+	 */
+	storageSync?: boolean;
+	/**
+	 * Override the default JSON.stringify function for custom serialization.
+	 * @param value
+	 */
+	stringify?: (value: unknown) => string;
+	/**
+	 * Override the default JSON.parse function for custom deserialization.
+	 * @param value
+	 */
+	parse?: (value: string) => unknown;
+};
+
+function isFunction(value: unknown): value is (...args: unknown[]) => unknown {
+	return typeof value === 'function';
+}
+
+function goodTry<T>(tryFn: () => T): T | undefined {
+	try {
+		return tryFn();
+	} catch {
+		return undefined;
+	}
+}
+
+function parseJSON(value: string): unknown {
+	return value === 'undefined' ? undefined : JSON.parse(value);
+}
+
+export const injectLocalStorage = <T>(
```

Comment:
nit: please use `assertInjector()` to ensure this function is being called in an Injection Context. Also, can we go with top-level `function` declaration instead of Arrow Fn expression?